client/src/lib/data.ts

```diff
@@ -13,3 +17,174 @@ export function readToken(): string {
   if (!token) throw new Error('No token found');
   return token;
 }
+
+export async function postSignUp(event: FormEvent<HTMLFormElement>) {
+  const formData = new FormData(event.currentTarget);
+  const userData = Object.fromEntries(formData);
+  const req = {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(userData),
+  };
+  const res = await fetch('/api/auth/sign-up', req);
+  if (!res.ok) {
+    throw new Error(`Fetch error: ${res.status}`);
+  }
+  const user = await res.json();
+  return user;
+}
+
+export async function verifySignIn(event: FormEvent<HTMLFormElement>) {
+  const formData = new FormData(event.currentTarget);
+  const userData = Object.fromEntries(formData);
+  const req = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(userData),
+  };
+  const resp = await fetch('/api/auth/sign-in', req);
+  if (!resp.ok) {
+    throw new Error(`${resp.status}`);
+  }
+  const payload = await resp.json();
+  return payload;
+}
+
+export async function getDetails(filmId: string | undefined) {
+  const detailsResp = await fetch(`/api/films/${filmId}`);
+  if (!detailsResp.ok) throw new Error('Failed to fetch film details');
+  const details = await detailsResp.json();
+  return details;
+}
+
+export async function getWishlist(
+  user: User | undefined,
+  filmId: string | undefined
+): Promise<boolean> {
+  if (user) {
+    const wishlistReq = {
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${readToken()}`,
+      },
+    };
+    const wishlistResp = await fetch(`/api/wishlists/${filmId}`, wishlistReq);
+    if (!wishlistResp.ok) throw new Error(`${wishlistResp.status}`);
+    const [isWishlist] = await wishlistResp.json();
+    if (isWishlist) {
```

Comment:
This construct is usually frowned upon. Instead, just use `return isWishlist;`. However, that doesn't always work because `isWishlist` may not be a boolean. To convert an arbitrary value to a boolean, you can use:
`return !!isWishlist;`. This works because the first `!` converts the expression into a boolean with the opposite truthiness of the original variable, and the second `!` puts it back to the original truthiness, but keeps it as a boolean.